### ISSUE TITLE
Django Suit version 2.0

### ISSUE THREAD

**Issue by darklow** [darklow/django-suit#475]
### Brace yourselves, this is going to be a long post that usual! :) 

I have bad news and good news.
The bad news is - Bootstrap 3 branch will be discontinued.
The good news is what is coming instead! 

### Some history

Some time after i launched the very first Django Suit (DS) version I realised biggest challenge is to keep up with Django admin changes. And `bs3` branch version proved that even more. I haven’t finished it yet as Django admin started changing quite a lot. New templates, new features, new styles etc. Third party apps are also changing and evolving a lot, therefore same problem existed with 3rd party apps too. I knew i needed to solve this issue before continuing.

### Flexbox

Recently I did a lot of thinking and experimenting and because browsers have evolved significantly I found an answer to my problems - Flexbox. I’ve been working on Django Suit fresh rewrite for few weeks and haven’t overridden a single Django admin template and in the end of the post you can see real screenshot of what i achieved by CSS only.

So this is a short story how Django Suit v2.0 rewrite started.

I will try to avoid any Django admin templates override as long as possible and if needed i will use JS DOM manipulations instead of overriding templates, which gives a few huge benefits:
When Django admin templates changes, some minor visual bugs may appear, but it will not break anything.

Without overriding admin templates, we have much wider 3rd party app support. Any app that haven’t modified templates too much and use Django admin markup should work perfectly.
This gives us much wider Django version support range, however I think Django Suit will stay with 2-3 recent version support.

### What’s new in Django Suit v2

* Based on [Bootstrap 4.0](http://v4-alpha.getbootstrap.com) (currently v4-alpha):
 * Moved from Less to Sass.
 * Flexbox usage and support.
 * Responsive sizing using `rem` units. To increase whole admin sizing (font-sizes, margins), all you have to do is change font-size for body tag.
* Django 1.9+
* Menu rewrite to `MenuItem` classes
* FontAwesome icons
* Demo app will be a part of Django Suit repository now
* Menu:
 * Horizontal navigation
 * Aligned right menu
 * Rewrite whole menu builder to MenuItem classes

So here is the current state of v2.0 rewrite.
No templates overridden except `base_site.html`:

![image](https://cloud.githubusercontent.com/assets/445304/12699480/3eee898e-c7c5-11e5-931c-ba1b0cabdecb.png)

Complete:
- [x] CSS: Base layout
- [x] CSS: List base layout
- [x] Menu builder rewrite

Todo:
- [ ] CSS: List states
- [ ] CSS: Forms
- [ ] CSS: Alerts & messages
- [ ] CSS: Widgets
- [ ] CSS: Footer
- [ ] CSS: Responsiveness
- [ ] CSS: Login
- [ ] Horizontal list filters
- [ ] Tabs
- [ ] Sortables

There are still a lot to do therefore this branch is not yet ready even for development.

Work in progress links:
v2 demo: http://v2.djangosuit.com/admin/
v2 branch: https://github.com/darklow/django-suit/tree/v2
v2 docs: http://django-suit.readthedocs.org/en/v2/

Feel free to share your thoughts.

**Comment by chris-erickson** [comment]
We use Suit on every project (paid!), and it looks like the future is going to be good. 🎩 Really appreciate the work you've done.

**Comment by fyaconiello** [comment]
Any ballpark when suit v2 is going to be ready? I have a couple paid django 1.x folks i wanted to upgrade to 1.9, but i'm afraid to break their admin.

**Comment by gamesbook** [comment]
I really hope we will see support for Django 1.8 as that is a long term release - that could be a deal--breaker.

**Comment by gamesbook** [comment]
Also hope to see the left-menu option retained as an alternative to the top menu - there was a lot of discussion about this before -- the fact remains that there is simply more horizontal screen space than vertical and avoiding "bar creep is very desirable.

**Comment by darklow** [comment]
@gamesbook Thanks for your feedback. I'll look into Django 1.8 support. As the stage is early, probably I can adjust it to make work for both. Supporting Django LTS release makes sense.
And I added horizontal menu to features list as well, it was planned, just forgot to mention in roadmap.

**Comment by SalahAdDin** [comment]
What's about left or right sidebar?
I want all previous suit versio features.

**Comment by gamesbook** [comment]
@darklow Thanks for the positive feedback.  This a great product that we hope to keep using and deploying further in future.  (I see 1.8LTS has planned support through to April 2018).

**Comment by darklow** [comment]
Day by day suit v2 is getting more and more mature.

**Status update:**
[x] List layout fixes
[x] Forms base
[x] Submit row
[x] Delete confimations

PS. You can see total progress in main issue description.

**Comment by darklow** [comment]
I know many of you will like these features that I just deployed ;)

- [x] **Two column change form with submit row on right** (with responsive fallback to bottom)
- [x] AutoSized textarea

You can see full list of updates in v2 branch commit log:
https://github.com/darklow/django-suit/commits/v2

v2 demo: 
http://v2.djangosuit.com/admin/demo/country/170/change/

![image](https://cloud.githubusercontent.com/assets/445304/13114625/f301e576-d59c-11e5-8290-0131349b6af6.png)

v2 demo: http://v2.djangosuit.com/admin/

**Comment by nchampsavoir** [comment]
This is really shaping up quite nicely. :)

2016-02-17 16:39 GMT+01:00 Kaspars Sprogis <notifications@github.com>:

**Comment by darklow** [comment]
If anyone wants to contribute to Django framework and help Django Suit as well, there is a great chance to do it. There is an incomplete pull request at Django repository with a super-valuable feature started, which needs some help to make it to the final line. Author don't have time to finish what he started, however he already did a great job and almost finished it.

Django: Added template-based widget rendering
https://code.djangoproject.com/ticket/15667

Pull request and author comments about what still has to be done to finish it:
https://github.com/django/django/pull/4848#issuecomment-173802002

I will try to help as much as possible, but i already have my hands full with v2 and other things.
If we could get this PR by Django version 1.10 then no more hacks for widgets are needed and anyone can style any widgets with custom templates. I think this is awesome and very very valuable feature.

Thanks.

**Comment by darklow** [comment]
Isn't this cute? :) 
Native Django admin Date/Time widgets:

![image](https://cloud.githubusercontent.com/assets/445304/13144158/8d5cac78-d651-11e5-8dc9-e68afb1371b1.png)

![image](https://cloud.githubusercontent.com/assets/445304/13144196/c078a90e-d651-11e5-8487-dff681331a31.png)

![image](https://cloud.githubusercontent.com/assets/445304/13144208/d2764850-d651-11e5-9769-7cafc9873da2.png)

Live demo: http://v2.djangosuit.com/admin/demo/showcase/add/

**Comment by platzhersh** [comment]
This looks great! good job!
Am 18.02.2016 14:11 schrieb "Kaspars Sprogis" <notifications@github.com>:

**Comment by krautnerds** [comment]
Awesome!

**Comment by darklow** [comment]
Some users with existing licenses asked if they can contribute or buy an upgrade fee for v2 in advance.
So even if this is complete rewrite, all you current licenses will be valid for v2 and therefore will be no upgrade fee. If you want to show your appreciation and support development of v2, you can use Donate feature or buy an extra license here:
http://djangosuit.com/pricing/#v2
Thank you!

**Comment by darklow** [comment]
Sortables for ``change list`` and ``tabular inlines`` are ready!
http://v2.djangosuit.com/admin/demo/continent/
http://v2.djangosuit.com/admin/demo/continent/9/change/

**Tips and Tricks**
By these years I've learned a lot of how to extend Django admin and in new v2 documentation I will share a lot of my knowledge how to do that too. 

Just added to demo app how to extend `submit_line.html`, `object-tools` or add custom admin actions for specific ModelAdmin:
http://v2.djangosuit.com/admin/demo/showcase/1/change/
https://github.com/darklow/django-suit/blob/v2/demo/demo/templatetags/demo_tags.py#L6
https://github.com/darklow/django-suit/blob/v2/demo/demo/admin.py#L190
https://github.com/darklow/django-suit/blob/v2/demo/demo/admin.py#L119

Hope you find it useful ;)

**Comment by SalahAdDin** [comment]
:+1:

**Comment by darklow** [comment]
Stacked inlines including Sortable Stacked inlines are now ready! 
http://v2.djangosuit.com/admin/demo/showcase/1/change/
![image](https://cloud.githubusercontent.com/assets/445304/13504763/92860b58-e17e-11e5-84c1-2c8374c022af.png)

**Comment by SalahAdDin** [comment]
:+1: I like it man!

**Comment by darklow** [comment]
Anyone using v2.0 alpha in development already or is it just me? :)

**Comment by otondin** [comment]
It's getting gorgeous! :+1:

**Comment by landaverdelbo** [comment]
me in mexico, for school management - testing...

**Comment by darklow** [comment]
Happy to announce next two handy features that are finished:
- [x] Form Tabs
- [x] Form Includes

Beta release is coming closer by each day!
http://v2.djangosuit.com/admin/demo/country/170/change/

![image](https://cloud.githubusercontent.com/assets/445304/13547192/48fc0566-e2d0-11e5-8644-ed336bebbfec.png)

![image](https://cloud.githubusercontent.com/assets/445304/13547195/625a2394-e2d0-11e5-857c-e25e90328af1.png)

**Comment by willseward** [comment]
This is awesome! Keep up the good work!

**Comment by Nekmo** [comment]
This looks good! :D

Thanks for your work.


Un cordial saludo:
-- Nekmo.

Sitio web: http://nekmo.com
Dirección de contacto: contacto@nekmo.com
XMPP/Jabber: contacto@nekmo.com
Google+: Nekmo Com

2016-03-09 18:14 GMT+01:00 Wills Ward <notifications@github.com>:

**Comment by mauriciobaeza** [comment]
Yes, it's a great software... but, please, we support their development
http://djangosuit.com/pricing/#v2

**Comment by SalahAdDin** [comment]
Eh, put a 400 error template please.

**Comment by oesah** [comment]
Awesome work, can't wait for V2!!!

**Comment by lukas-schulze** [comment]
Do you plan to support a left-sided navigation like in v1?
Most monitors have more space in the width than height and I would really prefer of move the navigation to the left instead of top. :-)

**Comment by eviltnan** [comment]
@lukas +1

**Comment by darklow** [comment]
@lukas-schulze I am thinking of left-sided navigation. But stick has two ends. Submit button column is also right sided. So by adding left side menu the space for form content is narrowing even more down. Users were complaining about that in v1. Maybe narrowing top navigation could be a solution as well.

**Comment by darklow** [comment]
@eviltnan Please use new Github +1 feature for comment or whole issue instead of a comment with +1.

**Comment by Sovetnikov** [comment]
This makes django admin enter to the new level!
Two usability features that will make life even better with Suit:
1. Drag'n'Drop for ordering (instead of up/down arrows)
2. New, redesigned filter_horizontal widget (as in groups and permissions for user changeform). At now it is very ugly and can be used only on small amounts of items available for select, it even lacks filtering in selected items list.

**Comment by gamesbook** [comment]
@lukas-schulze This topic was raised earlier in this issue and @darklow said he would support it.

Apart from the very good reason of it maximising available screen use, its also important for maintaining continuity with existing deployments; end users get a bit cranky if you "improve" things without really solid justification.

**Comment by joseph2002** [comment]
the package installed via 
pip install https://github.com/darklow/django-suit/tarball/v2
does not include bootstrap css?
How can I deal with this?

**Comment by darklow** [comment]
@joseph2002 Django Suit v2 doesn't have separate `bootstrap.css` file. It is all compiles using `sass` into single `suit.css` file: https://github.com/darklow/django-suit/tree/v2/suit/static/suit/css

**Comment by joseph2002** [comment]
@darklow It's ok now after I cleared my browser's cache. before that I can not see any styles, just plain texts. maybe something conflict with previous version of suit. thank you!

**Comment by darklow** [comment]
If you're already using `v2` or playing with it, until there are no official docs, read my comments here if you're interested in how to control form sizing in `v2`:
https://github.com/darklow/django-suit/issues/500

**Comment by kcolton** [comment]
@darklow  👍 We've been happy suit 1.0 users, then happy suit bs3 branch users, and look forward to being happy suit 2.0 users. Donated as a small token of our appreciation. Cheers 🍻 

http://djangosuit.com/pricing/#v2

**Comment by darklow** [comment]
@kcolton Thank you, I appreciate it. Next week I have few important deadlines and after that I am planning to finish things that are still missing and release v2 beta and put it on pypi.

**Comment by darklow** [comment]
For those who are using v2, just added missing style for two-select/column selector widget which is used for permissions:
http://v2.djangosuit.com/admin/auth/group/add/

**Comment by luto** [comment]
Just donated to your awesome project to show our thanks. Keep up the good work on v2! :+1:

**Comment by darklow** [comment]
@luto Maybe you donated to somebody, but definitely not me or DjangoSuit :) Thanks for good intentions, though! :)

**Comment by luto** [comment]
well, my boss donated a few weeks ago, by buying a license :) "just" was maybe the wrong word :sweat_smile:

**Comment by darklow** [comment]
Another awesome feature migrated:
Added `suit_row_attributes`, `suit_cell_attributes` callables for row/cell/column styling based on object and field land added one new `suit_column_attributes` callable for column attributes:

Demo:
http://v2.djangosuit.com/admin/demo/continent/
https://github.com/darklow/django-suit/blob/v2/demo/demo/admin.py#L140

Preview:
![image](https://cloud.githubusercontent.com/assets/445304/16013686/1977d07a-3197-11e6-8004-61b1b1fadcd7.png)

**Comment by henriquechehad** [comment]
👍

**Comment by uniqe15** [comment]
How about inline in inline? it is very necessary for many projects

**Comment by darklow** [comment]
@uniqe15 Inline in inline is way too complicated, there are few apps that does that, I tried, but they are always outdated and have many bugs and limitations.

However I have implemented in on of my projects "kind of" inline in inline solution, where you can edit 2nd level inline items in a popup. This is 100% valid solution using only Django admin templates to achieve this. I will include in the demo app to show how it could be done. Here is a preview:
![image](https://cloud.githubusercontent.com/assets/445304/16759502/e7b82e50-481f-11e6-8d14-6e6dc490795e.png)

**Comment by uniqe15** [comment]
I cannot wait to try it out! Is there any official release date?

**Comment by darklow** [comment]
@uniqe15 No official release date yet, but I am using v2-dev already in 3 projects in production.

**Comment by uniqe15** [comment]
I'm writing a project for graduation (poll app) and i have a lot of inline-in-inline stuff, is there a way to clone this v2-dev package? I'm not going to use it in production ofc but it will make my work muuch easier.

**Comment by darklow** [comment]
@uniqe15 In my first post in this issue I wrote how to install v2-dev: https://github.com/darklow/django-suit/issues/475#issue-130059324 however there is no example for inlines in inlines included yet.

**Comment by geekashu** [comment]
@darklow It seems django-debug-toolbar is not working with v2. Would appreciate if you could please check at your end.

**Comment by darklow** [comment]
@geekashu It is working but is getting hidden. It bootstrap fault,  this line hides it https://github.com/twbs/bootstrap/blob/v4-dev/scss/_reboot.scss#L391
I will look forward to override it somehow.

**Comment by darklow** [comment]
@geekashu I just fixed Django Debug Toolbar problem, just reinstall v2-dev:
```
pip uninstall django-suit
pip install https://github.com/darklow/django-suit/tarball/v2
```

**Comment by AliPolat** [comment]
thank you,.. hoping for nested inline demo soon..

**Comment by geekashu** [comment]
yup it works now... thanku...

**Comment by onyekaa** [comment]
@darklow Weird issue I'm having where the horizontal menu renders in development but is only an empty `ul` in production. Same settings. Any ideas?

**Comment by mzbyszewska** [comment]
I' using Django Suit v2. Is it possible to show Suit menu at frontend?

**Comment by darklow** [comment]
@mzbyszewska I don't think so, at least not in an easy way, Django Suit is "skin/addon" specifically for Django admin app and highly relies on it's context variables.

**Comment by mzbyszewska** [comment]
It seems that I should slightly overwritte your get_menu tag in case "if not available_apps" and fake available_apps for frontend. Then the MenuManager will has chance to read custom SuitConfig.

**Comment by geekashu** [comment]
I am using custom formset for my inline admin, inline formset are doing some custom validation and raises a `ValidationError` if certain criteria doesn't met for all the forms.

Now it seems triggered error messages are not able to use `errorlist` css class from `suit.css` and renders them as plain black colored text.

**Comment by darklow** [comment]
@geekashu please open new issue and provide a screenshot, I can't reproduce.

**Comment by dwasyl** [comment]
@darklow this update looks great, just looking into trying this out for the first time. That said, nested inlines is a huge feature for a few projects of mine and it would be great to see support for it. At the moment, I've used `dj-nested-inlines` before, and it seems that `django-nested-admin` has support for the current version of `django-suit`. It would be great to see this continue through a 3rd party module or just be builtin.

While there aren't necessarily a ton of use-cases there are a few where it's pretty indispensable.

**Comment by darklow** [comment]
Note about v2 support for Django 1.8 (LTE) @gamesbook.

For a few days I've been trying to add support for Django 1.8 and it isn't so easy and I am considering not adding it for upcoming v2. Although 1.8 is long-term release it is missing some important features which made Django Suit v2 so easy and clean to develop. 

The thing is in Django 1.9+ template loader was updated and improved which made possible to extend Django admin templates in very simple and clean way without overriding whole `change_form.html` and other main Django admin templates files, but just few parts of it.

In order to support Django 1.8 I would need to go back to fully copy/pasting full `change_form.html` and other file content and making appropriate changes, which would make problems supporting 3rd party apps "out-of-the-box" and other issues.

Current Django `v2-dev` already supports Django 1.9 and Django 1.10.

I think I must look forward this time and not breaking nice and simple things I created to support Django 1.8 for which mainstream support has already ended and only security fixes will be added.
https://www.djangoproject.com/download/

By the way Django 1.10 which is already out has support until December 2017 (just 4x months difference to Django 1.8)

For those who need Django 1.8 version support, let me know what you think.

**Comment by darklow** [comment]
@chris-erickson, @gamesbook good news!

Forget my previous post, persistent trying and tests resulted in success and I found a way to support all three Django 1.8, 1.9 and 1.10 versions. 
I merged in`v2-django18` branch into https://github.com/darklow/django-suit/tree/v2 and now you can try it out.

Django 1.8 still have some cosmetic issues and some legacy css classes that need some attention, but so far it works.

Feel free to test it with Django 1.8:
```
pip uninstall django-suit
pip install https://github.com/darklow/django-suit/tarball/v2
```

**Comment by gamesbook** [comment]
That is good news. We are scheduled to update to 1.8 soon and would like to
sync this change with Suitv2.

**Comment by keyvanm** [comment]
suit.js:199 Uncaught ReferenceError: Suit is not defined

**Comment by darklow** [comment]
@keyvanm Please open separate issue with more details, Django version etc.

**Comment by keyvanm** [comment]
#546

**Comment by heathernicoale** [comment]
Love V2 so far! Convinced my supervisors to user for our development project. Question: are drop down lists for filters feasible for this version? Instead of or in place of the list_filter default?

**Comment by darklow** [comment]
@heathernicoale Yes, I will reimplement them as well for v2 too.
Just added to feature list.

**Comment by heathernicoale** [comment]
@darklow - I see. I overlooked that, I apologize! Can't wait! :)

**Comment by heathernicoale** [comment]
Just out of curiosity - do you have a general idea of the release date for V2? :) This is such a great product so far! Can't wait to see the finished product! :)

**Comment by ganmedia** [comment]
Can you explain how to install version 2 on Digital Ocean Django One-Click Install Image?
I have been trying using your tutorial but it never works.

Thanks.

**Comment by darklow** [comment]
@ganmedia I am not familiar with Digital Ocean and have no idea what One-Click install means.
To install django-suit v2 you need to run:
```
pip install https://github.com/darklow/django-suit/tarball/v2
```
Or if you're using requirements.txt file then use following line (keeping space as first char):
```
 -e git://github.com/darklow/django-suit.git@v2#egg=django-suit
```

**Comment by ganmedia** [comment]
I did it but  I got always the same error:

TemplateSyntaxError at /admin/
'suit_tags' is not a valid tag library: ImportError raised loading suit.templatetags.suit_tags: No module named apps

**Comment by darklow** [comment]
@ganmedia please open a separate issue with some more details, full stacktrace, Django & python versions etc.

**Comment by darklow** [comment]
Two new features just added to `v2-dev`:
- [x] Django Suit now supports multiple admin instances - you can define settings / menus per each admin_site.
- [x] Define `def menu_handler(items, request, context): return items` method to config class to manipulate menu items right before they are being rendered.

**Comment by gamesbook** [comment]
Hoping that we'll see an "official" release before year end - this would be a fantastic upgrade to offer clients for next year!

**Comment by darklow** [comment]
@gamesbook me too :) Funny or not one thing that holds me back for releasing something like public beta is not having done left side navigation. Since I like top/horizontal navigation much more I wouldn't use left side anymore. Hard to make/do something when you don't need it. You will never run out of vertical space, but you will run out of horizontal eventually and left side nav will help you do that even faster and then I got tens of issues about "my inlines doesn't fit in horizontally and wraps" kind of issues.

**Comment by SanderTuit** [comment]
@darklow Have you considered collapsible side navigation? It seems like that would work great on mobile and smaller screens: that way it wouldn't take up horizontal space if there isn't enough space.

**Comment by darklow** [comment]
@SanderTuit Yes I have, and that was my intention, but as you know for responsiveness you need extra time to handle all the layout changes, collapses etc. But in terms of essential features it wasn't in my priority list so far. First I wanted to deal with features that doesn't have alternative solutions.

**Comment by gamesbook** [comment]
@darklow In principle, you are right - infinite scrolling.  But the trick is to fit the key data into limited vertical space **without** users having to scroll.  I limit the default view to 10 or 15 rows for that reason.  An optional "hider" on the sidebar (like bitbucket uses) also makes sense.

**Comment by darklow** [comment]
Had quite a productive day.
- [x] Updated to latest Bootstrap v4-alpha5.
- [x] Vertical menu layout. (work still in progress and polishing details, but base is there).

Todo:
- [ ] Icon support.
- [ ] Vertical menu responsiveness.

Suggestions and feedback are welcomed.
Feel free to test it from `v2-vertical` branch:
```
pip uninstall django-suit
pip install https://github.com/darklow/django-suit/tarball/v2-vertical
```

```py
#apps.py
class SuitConfig(DjangoSuitConfig):
    layout = 'vertical'
```

![image](https://cloud.githubusercontent.com/assets/445304/20239862/5456d364-a913-11e6-8590-1cfbf6690a35.png)

**Comment by gamesbook** [comment]
Very happy to see "Return of the Left Menu".  Will the location of the little icons (between the title and the menu itself) be repositionable?  In the current version, the menu starts directly below top name; and we need as much vertical space as possible.

**Comment by ganmedia** [comment]
Great news, but where should I place apps.py to override SuitConfig?

**Comment by darklow** [comment]
@ganmedia You can put `apps.py` anywhere you want, but by Django documentation it should be in your main project main app directory.

This is the new way of configuring 3rd party apps which was introduced in Django 1.7 already:
https://docs.djangoproject.com/en/1.10/ref/applications/

See example in `demo/demo` app:
https://github.com/darklow/django-suit/blob/v2/demo/demo/apps.py
https://github.com/darklow/django-suit/blob/v2/demo/demo/settings.py#L39

**Comment by darklow** [comment]
@gamesbook Currently both horizontal and vertical layouts use same template and CSS is what puts them left or top. To manipulate position of `.header .user-tools` template you could override `base-site.html` although not too much places elsewhere where you could put them. I really didn't wanted to introduce another header inside center column as it was in v1.
Another option is - you can just override CSS and make these icons position: absolute and move them to the bottom of the left column, would that work to you?

**Comment by gamesbook** [comment]
@darklow I think bottom of the menu (left column) would be fine; but I
should also get feedback from some of the users.

**Comment by heathernicoale** [comment]
@darklow - this is shaping up great so far! :) Question though - I followed the uninstall, install instructions, however the vertical menu isn't display even after restarting the server. Any ideas what may cause it to not display? It's also been added to the SuitConfig.

**Comment by ganmedia** [comment]
@heathernicoale In settings.py add this line to INSTALLED_APPS:
`'django_project.apps.SuitConfig'`

Then add a file called apps.py in the root of your project directory (same where settings.py is) and add this:
```
from suit.apps import DjangoSuitConfig
class SuitConfig(DjangoSuitConfig):
    layout = 'vertical'
```

**Comment by heathernicoale** [comment]
@ganmedia Hey! :) Thanks for the response! I have done this. I had my menu already configured and set up. so all of the parent / child items are in place already. Everything is there, the apps.py, DjangoSuitConfig, InstalledApps, etc...

**Comment by darklow** [comment]
@heathernicolae read carefully my commet, v3 vertical is on another branch: https://github.com/darklow/django-suit/issues/475#issuecomment-260138558

**Comment by Nofcale** [comment]
Hi @darklow is really great the job you have done. I have been using suit v2 for almost 4 months and it has been outstanding. I was expecting the left menu and now I will wait the responsiveness feature. I want to use it in a recent project I am working on. Just wanted to thank you.

**Comment by AliPolat** [comment]
I just begin to use v2. The show_change_link = True  option in TabularInline does not show the change link. is it not implemented yet or am I missing something ?

**Comment by AliPolat** [comment]
Thank you for this beautiful admin interface.

I could'nt make work 'show_change_link' option of InlineModelAdmin .

**Comment by darklow** [comment]
@AliPolat just fixed in e997f71a165356dca1f4bd75b202af8767106a9c
Now there is global config `form_inlines_hide_original = False` for `SuitConfig` configuration class which may be overridden per individual Inlines as `suit_form_inlines_hide_original = True`
Demo with `show_change_link`: http://v2.djangosuit.com/admin/demo/continent/7/change/

**Comment by AliPolat** [comment]
Thank you, It has very nice appearance...

**Comment by AliPolat** [comment]
Hi darklow,

Thank you for the great work, What about responsiveness, will it be ready in near future ?

**Comment by heathernicoale** [comment]
@darklow - absolutely LOVE Django Suit V2 - your development has significantly progressed our own development on our company's project. 

Question - have you considered adding 'Print' actions or ability in Django Suit V2? I'm surprised this isn't already included in Django Admin anyhow. Working on creating a custom one for our project. But it made me think to ask. :) Keep up the great work!

**Comment by darklow** [comment]
@heathernicoale Thanks, glad you like v2.
I don't think many users need visible Print actions and print actions are always available from browser menus.
However having print styles (hiding redundant menus) to improve printing of lists and forms is a good idea. I'll add it to the feature list.

**Comment by darklow** [comment]
`v2-vertical` branch has been merged into `v2`.

**Comment by darklow** [comment]
@gamesbook Is Django 1.8 support really so necessary for you?
I keep fighting issues with Django 1.8 and circular imports because of outdated template loaders, which were fixed and rewritten in Django 1.9.

**Comment by gamesbook** [comment]
@darklow At this rate (no official release of v2 as yet), we can delay upgrading until April 2017, which is when Django 1.11LTS comes out.  Can I assume Suit will be supported in that release?

**Comment by darklow** [comment]
@gamesbook I was thinking more of - are you really considering moving your existing clients who uses Django 1.9 to suit v2 as v1 and v2 are not exactly compatible, some config and other changes will be required. 

I will try my best to release v2 this year. Instead, today I spent 4 hours just trying to deal with Django 1.8 compatibility and still haven't found good solution that nicely would work for 1.8 and 1.9/1.10. I really don't want to use some custom template tag for all my templates to fix circular imports for Django 1.8.

**Comment by kcolton** [comment]
@gamesbook This is going to come across rude, but it comes entirely from my own history of mistakes (as i've gotten stuck in dependency hell and fallen behind on being able to upgrade important frameworks and know its also often completely out of ones control). 

Without knowing anything about your situation, FWIW if you are stuck on 1.8, Django Admin being BS4 should probably not be high on the priority list. If you need Bootstrap, I would actually recommend the no longer maintained BS3 branch of suit. We used it in production for quite a while and it worked basically fine. 

IMNHO I think BS4 itself is suffering from it's own "backwards compatability" issues. By the time they get to an RC, flexbox will be so universally supported it wont matter that it has table support.

BS3 was a big step forward in that it just said screw it, `* { box-sizing: content-box; }` and any of the poor souls who needed to support older IE was "stuck" with a still perfectly usable BS2.

1.8 is was labeled LTS, and they have a duty to support that, but third party packages don't have the same support ecosystem and you can always pin older versions of packages.

Unfortunately, pip is not so magical to work the other way around, otherwise @darklow could just `pip install django-suit==2.0.0` and there would be no problem. 😆

**Comment by heathernicoale** [comment]
@darklow - thanks! Great idea! :) I had just been reading up on print styles last week. The only reason I had been looking into print options (or actions for that matter) was because we have so many rows of data. We can set up filters, but it still exceeds the visible print view and the pagination is also playing a factor in that as well. I was trying to add a print action / function to print ALL the rows in a dataset (even if it's not all visible in the browser window) -based on the filters or ordering set. Of course there may be another way to do this - I'm still a bit of a rookie. Thanks!

**Comment by gamesbook** [comment]
Our plan was to upgrade to 1.8LTS, but we need to keep using Django Suit (everyone we show it to reacts very well to the "look and feel") so I was waiting for v2 to support it before running any upgrades.  A DSv2/ Django1.11(LTS) is just as valid from our point of view.  The only issues might be - as @kcolton points out - *other* third party packages; but I am hoping they will also move to support 1.11, otherwise we will be forced to use a 1.9 or 1.10 in the interim.

I'd rather vote for an earlier release of v2 without 1.8 support than see it delayed (assuming my vote counts for anything).

**Comment by chris-erickson** [comment]
We have a lot of 1.8 LTS sites (and will until 1.11) so I would love to see 1.8 well supported.  That said, I can appreciate the difficulty doing so, and desire to get better prepared for the next LTS first.  We're I to vote, it would probably be to defer 1.8 support, so that the community as a while is better served - v1 works just fine for now.

**Comment by darklow** [comment]
@gamesbook @chris-erickson @kcolton Thank you for your feedback on Django 1.8. Since it was broke and I wasn't able to fix it, I am removing partial Django 1.8 support for now and will try one more time when v2 is finished.

**Comment by darklow** [comment]
Current login form seemed little too geeky and was bothering me for a long time. Because first impression is important and login form is first thing that your (and my) clients see, I did some work on it. Live demo:  http://v2.djangosuit.com/admin/
PS. Graphic is included only on demo, by default there is none. Example how to add one is here: https://github.com/darklow/django-suit/blob/v2/demo/demo/templates/admin/login.html#L4 
![image](https://cloud.githubusercontent.com/assets/445304/21441847/ee29e2ca-c8a2-11e6-849b-43d0cc68a180.png)
vs old:
![image](https://cloud.githubusercontent.com/assets/445304/21441821/cd409fcc-c8a2-11e6-9d28-e8c528b6733e.png)

**Comment by SalahAdDin** [comment]
New login is better.

**Comment by darklow** [comment]
@gamesbook @kcolton and others. What are 3rd party packages you definitely require to work with Django Suit v2? 

I would like to test them out first, how well they work out of the box, before releasing first alphas on v2. My intention from the very beginning was override as less templates as possible and use CSS/JS as first solution. This approach should allow other 3rd party packages to be supported without extra packages or too much work.

**Comment by gamesbook** [comment]
@darklow 

My list (by the INSTALLED_APPS name and project URL):

- 'crispy_forms': https://github.com/maraujop/django-crispy-forms
- 'autocomplete_light': https://github.com/yourlabs/django-autocomplete-light
- 'admin_enhancer': https://github.com/charettes/django-admin-enhancer (unneeded in 1.8+)
- 'smart_selects': https://github.com/digi604/django-smart-selects
- 'selectable': https://github.com/mlavin/django-selectable
- 'django_pandas': https://github.com/chrisdev/django-pandas
- 'massadmin': https://github.com/burke-software/django-mass-edit
    
Plus https://github.com/django-debug-toolbar/django-debug-toolbar is used for development.

**Comment by raratiru** [comment]
Thank you for this great effort. 

I would like to mention an old issue ( https://github.com/darklow/django-suit/issues/367 ) regarding[ HiddenInput widget](https://docs.djangoproject.com/en/dev/ref/forms/widgets/#hiddeninput) just to make sure that it will be fixed.

A very useful app, that I hope it will succeed in staying current and up-to-date, is [django-leaflet](https://github.com/makinacorpus/django-leaflet).

**Comment by hebijiandai** [comment]
Excellent!

**Comment by darklow** [comment]
To those who are using v2 and `suit_row_attributes()` be aware of backwards incompatible change:
https://github.com/darklow/django-suit/commit/ea9731b43a65d7f3eef7abb19d971e594be3b4d5

`request` will be always passed to `suit_row_attributes(self, obj, request)` as second argument.

```py
`suit_row_attributes()`
class MyAdmin(ModelAdmin):
    ...
    def suit_row_attributes(self, obj, request):
        return {'class': 'type-%s' % obj.type}
```

**Comment by a-trost** [comment]
Been loving this update. Is there a way to change the "Suit Admin" header at the top left to an image instead?

**Comment by darklow** [comment]
@a-trost Sure, extend or replace `{% block branding %}` in your `base_site.html` file.

**Comment by jon1924** [comment]
I've been experimenting with this beautiful admin (_v2_), and am considering using in production (the tab functionality alone would be a huge improvement for us!). I'm wondering whether you have any more info on a few things:
- horizontal `list_filter`
- `show_change_link` for inlines
- ability to use with _django-controlcenter_ (which I think extends `admin/base_site.html`)
- PyPi release (makes it easier to use with Docker)

If horizontal `list_filter` and `show_change_link` are needed, are you recommending people stick with _v1_ for now?
Thanks!

**Comment by darklow** [comment]
1. Horizontal list filters are definitely next on the todo list. I would say maximum in a week.
2. `show_change_link` is now natively supported by Django admin and also supported by Django Suit. See example here, you'll notice Change link: http://v2.djangosuit.com/admin/demo/continent/9/change/
I will need to do more research about `django-controlcenter` and pypi release.
For new projects or the ones under development I suggest using v2, for existing ones stick with v1 until all features are migrated.

**Comment by gamesbook** [comment]
Assuming you have jumped through the hoops to add packing info to your project, then this is a good way to test PyPI:
https://wiki.python.org/moin/TestPyPI

**Comment by darklow** [comment]
Oh, you meant deploying v2 on pypi, yes of course I'll do that, although I wanted to do that only when the new docs are online, to avoid confusion. I bet lot of users will blindly run update and wonder what happened.

Right now I am deploying my Docker apps following way:
```
# requirements.txt
Django==1.10.5
# Django suit invalidate: 1
 -e git://github.com/darklow/django-suit.git@v2#egg=django-suit
```

And when I need to upgrade Django Suit I just change invalidate number.
I know this is not perfect, but works as a workaround until pypi is online.

**Comment by jon1924** [comment]
Thanks for the fast reponse. That's good news re: `list_filter` and `show_change_link` (I must be overriding something with the latter).

Docker pulls in _django-suit v2_ fine - it's just that the container rebuilds every time there's a repo change, unless you tie it to a particular commit. A stable PyPI release version would just let the container stay calmer. I'll try your `invalidate` method - that's a bit more convenient.

Getting excited about using this!

**Comment by willseward** [comment]
@jon1924 Strange that the docker container rebuilds when there is an upstream change to the git repo. It should only be looking at the local filesystem when deciding to invalidate the cache.

Have you tried something like this?:

**Comment by willseward** [comment]
@jon1924 Strange that the docker container rebuilds when there is an upstream change to the git repo. It should only be looking at the local filesystem when deciding to invalidate the cache.

Have you tried something like this?:

```txt
COPY ./requirements.txt /src
RUN pip install -r /src/requirements.txt
COPY ./ /src
```

This will ensure that you can maintain a cached layer for the python installation and then modify your source code without invalidating the pip install. @darklow's invalidation solution is a good workaround until a pypi version is released.

**Comment by darklow** [comment]
@gamesbook @jon1924 @Sovetnikov Great news – horizontal list filters are ready! Even better - in classic v1 all filters where converted into horizontal, but now in v2 you can have both django original vertical filters and horizontal filters together. All you need to do is besides `list_filter` define also `suit_list_filter_horizontal` list for the vertical ones.

Live demo: http://v2.djangosuit.com/admin/demo/country/
Code example: https://github.com/darklow/django-suit/blob/v2/demo/demo/admin.py#L71

![image](https://cloud.githubusercontent.com/assets/445304/22856518/88ac584e-f09b-11e6-8465-9527a870737e.png)

**Comment by darklow** [comment]
In v2 instead of adding support for 3rd party packages I will include examples of usages, widgets etc in the demo app instead, so they can be used as an example to copy into your app when needed. The cons are that it won't break your app if you upgrade Django Suit or update 3rd party app and you get some version mismatch, in such case you can easily adapt widget and maintain code for you needs.

Because https://github.com/applegrew/django-select2 is one of the most popular packages, I added example of how to use Django-Select2 Model widget in v2-dev.

Demo: http://v2.djangosuit.com/admin/demo/showcase/1/change/ (Under Foreign key relations)
Code: https://github.com/darklow/django-suit/blob/v2/demo/demo/widgets.py#L6
https://github.com/darklow/django-suit/blob/v2/demo/demo/admin.py#L208

Related issues on Django-Select2 repo: 
https://github.com/applegrew/django-select2/issues/279
https://github.com/applegrew/django-select2/issues/284

I think I am ready to start writing new documentation for v2 now and then 🎉

**Comment by gamesbook** [comment]
@darklow "The cons are that it won't break your app" - you mean, "The pros
are that it won't break your app"?

Great news on horizontal filters - they are a must when the lists get too
long!

**Comment by darklow** [comment]
@gamesbook yes I meant pros :) thanks

**Comment by fastdevops** [comment]
@register.filter
def suit_list_filter_horizontal(filters, cl):
    filter_horizontal = getattr(cl.model_admin, 'suit_list_filter_horizontal', [])
    return [f for f in filters if f.field_path in filter_horizontal] 

When I define the custom admin filter extend the SimpleListFilter, the django-suit v2 throws the below exception
'StatusFilterForAdministrator' object has no attribute 'field_path'

**Comment by luison** [comment]
Just to inform that V2 demo seems to be down.
Happy to see v2 is being pushed but just trying to see what is the current situation with v 1.9 support. I've read that 1.8 support is not 100% and seen a requirements.txt with 1.10. Is that going to be the minimum supported version initially?
Thanks.

**Comment by gamesbook** [comment]
Given that Django 1.9 extended support stops in 2 months time, it does not seem to make sense to try & support it (for small projects, at least).

**Comment by darklow** [comment]
@luison and @gamesbook Thanks for letting know v2 demo is down, wasn't aware of that. Now it is back again, was playing with docker and accidentally mess it up a little.
Currently Django Suit supports perfectly both Django 1.9 and Django 1.10, until it doesn't make hard to maintain, can't see why to drop support for Django 1.9 either. @luison `requirements.txt` is just for the demo. Also non of my current projects had any problems moving from 1.9 to 1.10, so I would say by my experience they are very compatible.

I am starting to doubt about wether to support Django 1.8, because of it's older template loader system, which makes it really hard extend in a clean way and avoid circular imports, also Django 1.11 is not so far away, so maybe I should look forward, not backwards.

**Comment by luison** [comment]
Thanks for the quick and detailed reply.
We´ll go ahead and try it on a project running on 1.9

**Comment by fastdevops** [comment]
If I use the static external URL in the menu, how to format the URL base on the request user language?

ChildItem('web site', url='http://test.site/{}/index.html'.format(language_code))

Thanks!

**Comment by gamesbook** [comment]
I cant speak for anyone else, but v.10 and v.11 support would be more than adequate.

**Comment by darklow** [comment]
@fastdevops url can be callable too. `request` and `context` variables will be passed:
https://github.com/darklow/django-suit/blob/v2/suit/menu.py#L217
PS. Please open separate issue if you want to discuss the issue more

**Comment by fastdevops** [comment]
ok, i understand it. 

The django suit version v2 whether support the django filer.

**Comment by alexphelps** [comment]
@darklow Firstly, I want to say that v2 is awesome. I just getting started with it on a new project. 

I would like to know how to move the Change Form Submit buttons down to the bottom like standard django for wider forms. I have quite large codemirror textarea form widgets and on a small screen with the navigation on the and submit row on the right which results in a slim middle column for the form. 

Would this be possible or possibly a setting to be able to toggle the change submit buttons location?

**Comment by darklow** [comment]
@alexphelps Thanks.
Yes there is a param to do that, set `form_submit_on_right  = False` in your `SuitConfig` class

You can see all available configuration parameters in DjangoSuitConfig base class while docs are under development: https://github.com/darklow/django-suit/blob/v2/suit/apps.py#L44

**Comment by alexphelps** [comment]
@darklow I just tried the setting. This is the result for me. 

![image](https://cloud.githubusercontent.com/assets/674282/23532169/b6a3e3c4-ff6f-11e6-9d3a-a91b8b1adb22.png)

**Comment by darklow** [comment]
@alexphelps Looks like I have overridden something while developing new features. Could you please open separate issue, I will take a look, meanwhile please use default settings.

**Comment by darklow** [comment]
`EnclosedInput` widget has been added:
![image](https://cloud.githubusercontent.com/assets/445304/23703738/19e375a0-040a-11e7-8fa0-de16e833e637.png)

Demo:
http://v2.djangosuit.com/admin/demo/country/170/change/
https://github.com/darklow/django-suit/blob/v2/demo/demo/admin.py#L48

**Comment by brunotikami** [comment]
hey @darklow! first of all, congrats on the outstanding work with v2. Do you have any thoughts on licensing for v2? Should I buy a v1 license, would it be valid or upgradable for v2? Thanks !

**Comment by darklow** [comment]
@brunotikami thanks. You'll find answers about licensing v2 here https://github.com/darklow/django-suit/issues/475#issuecomment-285850484 and here http://djangosuit.com/pricing/#v2

**Comment by darklow** [comment]
Another new feature added to v2 - `RelatedFieldAdmin` - improved ModelAdmin class that can show any related/FK field as link or text and is not limited by lookup depth (Django supports only direct FK fields):

Usage: Just add `link_to_` prefix to any related field.
```
from suit.admin import RelatedFieldAdmin

class ShowcaseAdmin(RelatedFieldAdmin):
    list_display = ('link_to_user', 'address__city', 'link_to_address__city')
```

Demo:
http://v2.djangosuit.com/admin/demo/country/
http://v2.djangosuit.com/admin/demo/showcase/

Overriding `short_description` or `admin_order_field` example:
https://github.com/darklow/django-suit/blob/v2/demo/demo/admin.py#L241

Screenshot:
![image](https://cloud.githubusercontent.com/assets/445304/24253557/553e94e6-0fe9-11e7-8207-fd8c8e06c78f.png)

**Comment by ebertti** [comment]
Hi @darklow, about this relataed link, I made this lib, without using mixin or inheritance

I think you don't need to a Base Class for this.

https://github.com/ebertti/django-admin-easy

```python
from django.contrib import admin
import easy

class ShowcaseAdmin(admin.ModelAdmin):
    list_display = ('link_user', 'address__city', 'link_address_city')

    link_user = easy.ForeignKeyAdminField('user')
    link_address_city = easy.ForeignKeyAdminField('address.city')

```

**Comment by darklow** [comment]
@ebertti Your library is quite nice indeed, I wasn't aware of it. I explored options using decorators, but wasn't sure it would be easy overriding `queryset` to add related fields to `select_related()` so I went with Inheritance which is classic way to extend things and I see nothing wrong with it.

I don't think I am ready to drop feature just yet, but I like how you factored this feature as a separate fields instead of using "magic" `link_to_` prefix pattern like I did, so I'll probably rewrite these two existing into something similar.

PS. You should use admin_site `current_app` variable when generating links to support multiple admin sites. See how Django does it: https://github.com/django/django/blob/master/django/contrib/admin/views/main.py#L386

If you wan't to discuss more about this, let's open separate issue.

**Comment by Nekmo** [comment]
Thank you very much for your work! :D We are already upgrading from SuitV1 to SuitV2 without problems.

The only thing we miss, and I have not seen in TODO list, is the search. Are you planning to retrieve the search box?

Thank you so much for everything!

**Comment by darklow** [comment]
Django 1.11 was released today and didn't noticed any issues with v2. Feel free to upgrade to Django 1.11 to get all the great new features: https://docs.djangoproject.com/en/1.11/releases/1.11/

**Comment by gasinvein** [comment]
Will there be some v1→v2 migration guide?

**Comment by gamesbook** [comment]
@darklow Can you confirm that suit2.0 will support Django 1.10 and 1.11 going forward?

**Comment by darklow** [comment]
@gamesbook Yes I can confirm that. Right now I am using v2 in different projects with Django 1.9, 1.10 and 1.11 with no issues so far. At one point in future maybe I will introduce some new widgets that are only supported since Django 1.11, because of the new template based widget system that was introduced in 1.11.

**Comment by SergSergiu** [comment]
When i try to add something , main menu always show first item/Model. I do something worng or is a bug?
![menu_error](https://cloud.githubusercontent.com/assets/9361205/24964606/6798255a-1faa-11e7-9985-41c842b1cf0f.png)

**Comment by gamesbook** [comment]
@SergSergiu  see #601

**Comment by larryhq** [comment]
When extend ImportExportMixin in django admin, export and import page style is error in suit2.0.
Will it be modify later?

**Comment by darklow** [comment]
@larryhq Please open a separate issue providing more details.

**Comment by madthew** [comment]
Just 2 quick questions:  there are 2 features I would really like to see in this new version not sure if there is any plan to develop thos:
- a nice date range filter with date from - date to
- possibility to hide/show columns and memorize the shown column for the specific user. 

With those 2 features I believe will be perfect!

**Comment by darklow** [comment]
@madthew 
1) Date range - I will consider. 
2) I don't feel such feature is needed as it is one of those hidden features that your users never use. The less things to configure for users, the easier and better is the UI. You should show and if you need some extra column, I usually just add checkbox to search row with label something like "Show photos" which toggles one or another column. Or sometimes I change columns based on filters that are used and when it makes more sense to show other columns, but I don't think user configurable columns is a good feature for end users. There was a PR once, so you may take a look and try to reimplement if you want it for your project: https://github.com/darklow/django-suit/pull/132

**Comment by SergSergiu** [comment]
@gamesbook , works! Thank you!

**Comment by jorenham** [comment]
Is there an ETA for the full v2 release?

**Comment by darklow** [comment]
@jorenham it is basically for 99% done and I'm using it on all my production projects, unfortunately I haven't got time to rewrite all the tests and docs to make it official.

**Comment by jorenham** [comment]
@darklow Thank you, that is good to hear! I will start integrating it in my production project in than case.

**Comment by heathernicoale** [comment]
@darklow - Hello! :) First off - LOVE V2. Excellent product. Hoping you can point me in the right direction... I've added Django Import/ Export - works well -except at one point when I go to export -the home screen doesn't have any formatting... I don't want to jack anything up - do you have any recommendations on what I need to do to maintain the Django suit theme within the Django Import/ Export app?

**Comment by deschler** [comment]
Do you see any chance to render/style the list filters as dropdown menus, like [django-grappelli](http://grappelliproject.com/) does? This is extremely useful when you have filters with many entries, where the page is stretched otherwise.

**Comment by gamesbook** [comment]
@deschler Not sure I get exactly what you are asking, but have you looked at `suit_list_filter_horizontal`?

**Comment by gamesbook** [comment]
@darklow Any simple example of how to change the site header?  In past version, it was the `ADMIN_NAME` in `SUIT_CONFIG`.  In the new `apps.py` file, I have tried this:
```
class SuitConfig(DjangoSuitConfig):
    layout = 'vertical'
    name = 'suit'
    verbose_name = 'mysite'
    admin_name = 'My Site'
```
But none makes any difference? 

(It would also be very helpful to show how the old settings map to their new equivalents, to allow us older users an easier migration path...)

**Comment by darklow** [comment]
@gamesbook While building v2 from scratch I realised there is Django way to override and I decided not to introduce new way of doing that. I'll add this to docs as well. So here is the Django way of changing header title for projects - In your main admin.py (or actually any of your `admin.py` files) do following:
```py
from django.contrib import admin

admin.site.site_header = 'MySiteName'
```

If you have multiple admin sites under one project, this is the way you define title properly for each of them.

**Comment by darklow** [comment]
@deschler There is such feature. Take a look here on horizontal filters in the demo:
http://v2.djangosuit.com/admin/demo/country/
Usage: https://github.com/darklow/django-suit/blob/v2/demo/demo/admin.py#L71

**Comment by gamesbook** [comment]
@darklow Thanks - the official docs describe it here: https://docs.djangoproject.com/en/1.11/ref/contrib/admin/#adminsite-attributes

**Comment by gamesbook** [comment]
@darklow For some reason, the admin messages are not showing up after an action has been completed; can you confirm they are still active/working (I have a 'vertical' layout)?

**Comment by darklow** [comment]
@gamesbook Yes, just tested demo and seems like a bug for vertical layout only. Could you please open separate issue, thanks

**Comment by deschler** [comment]
@gamesbook Thanks for pointing out  `suit_list_filter_horizontal`!

**Comment by gamesbook** [comment]
@darklow 
_So here is the Django way of changing header title for projects - In your main admin.py (or actually any of your admin.py files) do following_
So far have been unable to get this feature to work - there is no "main" `admin.py` file for a Django project, and adding this to others makes no difference.  Is there an example in the demo code?

**Comment by jorenham** [comment]
@gamesbook https://github.com/darklow/django-suit/blob/v2/demo/demo/admin.py#L19

**Comment by gamesbook** [comment]
@darklow Thanks, I will try this. I assume it makes no difference which app it appears in?

**Comment by aditya1315** [comment]
I want to use version 2 of django-suit. But I get this error, 
`    from suit.widgets import SuitDateWidget
ImportError: cannot import name SuitDateWidget`

What is the alternative for SuitDateWidget in v2?

**Comment by darklow** [comment]
@aditya1315 there is no alternatives to these classes in v2 as there is no need for these anymore, I have styled them using other approaches, you can use native classes (don't override at all).

**Comment by gamesbook** [comment]
@darklow is there any guide on how to retheme suit?  I have a client that would like to see "shades of green" instead of the current greys, in the interface.  So, nothing fancy, just replacing some of the main colors used.

**Comment by darklow** [comment]
@gamesbook It is quite easy actually, here are some docs (See section Compiling SASS):
https://github.com/darklow/django-suit/blob/v2/docs/contribute.rst

Here are all the base colours defined:
https://github.com/darklow/django-suit/blob/v2/suit/sass/_variables.scss

**Comment by gamesbook** [comment]
@darklow Thanks - should I define my own `custom.scss` file which overrides some of these and add that file to my project's CSS folder?

**Comment by darklow** [comment]
@gamesbook It is up to you how you want to structure customised version, but end result will still be one `suit.css` file for you client project, it will be quite hard with overriding by extra css file. If you have more questions about extending v2, let's open separate issue. Thanks.

**Comment by prince-tanapong** [comment]
@darklow  Do you have a plan to implement this feature `Warning on leaving if unconfirmed changes`?  I think it is the important one for prevent user exit when they have some changed.

Thanks

**Comment by gamesbook** [comment]
@darklow see #629

**Comment by madthew** [comment]
Hey, guys I'm not getting something...  I'm defining my config as the following and those models exists but someone the menu don't show up the link... it just shows custom. I don't know what I'm duing wrong.

```
class SuitConfig(DjangoSuitConfig):
    layout = 'vertical'
    menu = (
        ParentItem('Content', children=[
            ChildItem(model='WebGateway.RuleGroup'),
            ChildItem('Custom view', url='/admin/custom/'),
        ], icon='fa fa-leaf'),
        ParentItem('Integrations', children=[
            ChildItem(model='demo.city'),
        ]),
        ParentItem('Users', children=[
            ChildItem(model='auth.user'),
            ChildItem('User groups', 'auth.group'),
        ], icon='fa fa-users'),
    )
```

**Comment by darklow** [comment]
@madthew Please post a separate issue as this is issue to discuss v2 features not specific issues:
1) When posting issue add content of your `settings.py`. 
2) Also try horizontal layout and see if it works there. direction in general.
3) Do you see non of menu items or just some? I see your are using CamelCase for `WebGateway.RuleGroup` you should use lowercase unless your `WebGateway` directory is actually called in such way.

**Comment by jorenham** [comment]
FYI Bootstrap 4.0.0-beta has been released

**Comment by jturmel** [comment]
Is there any sort of target date for having responsiveness complete? We went ahead and bought a paid license because we enjoy this project, but one of the things we're most needing is v2 in responsive.

**Comment by mhindery** [comment]
The list filters which in the current version show as dropdowns are now shown all in the right column with all options expanded. Will there be a setting to have these as dropdowns on top? From a usability perspective, this change is quite a step backwards when having multiple filters, especially with long lists of choices (e.g. country codes, currencies ...).

**Comment by darklow** [comment]
@mhindery Take a look here on horizontal filters in the demo:
http://v2.djangosuit.com/admin/demo/country/
Usage: https://github.com/darklow/django-suit/blob/v2/demo/demo/admin.py#L71

**Comment by gamesbook** [comment]
@darklow It seems a few people missed this feature - perhaps worth adding to the docs (apologies if it is already there).

**Comment by ganmedia** [comment]
Can I use CKEditorWidget or RedactorWidget in v2?
I tried following the documentation but WYSIWYG editor is not loading.

This is the documentation I used: http://django-suit.readthedocs.io/en/develop/wysiwyg.html

**Comment by JacoBezuidenhout** [comment]
Hi @ganmedia. I am using django-tinymce in my project that runs on Suit V2. Maybe this helps... in my models.py

`from tinymce.models import HTMLField`
....
`    gp_temp = models.CharField(max_length=255, null=True, blank=True)`
`    medical_note = models.TextField(null=True, blank=True)`
-- > `    note = HTMLField(null=True, blank=True)`
`    dependant_code = models.CharField(max_length=2, null=True, blank=True, choices=(`
`        ('00', 'Main Member'),`
`        ('01', 'Spouse'),`
....

**Comment by larryhq** [comment]
Hello, I want ask a question about list_filter and suit_list_filter_horizontal,
when I custom a filter extent SimpleListFilter,like this
```
from django.contrib.admin.filters import SimpleListFilter

from ..projects.models import Project
from .models import Order


class ProjectFilter(SimpleListFilter):
    title = '所属项目'
    parameter_name = '所属项目'

    def lookups(self, request, model_admin):
        project_list = [item.project_id for item in Order.objects.values('project_id')]
        choices_list = [(0, '----')]
        for project in Project.objects.filter(id__in=project_list):
            choices_list.append((project.id, project.name))
        return choices_list

    def queryset(self, request, queryset):
        qs = Order.objects.filter(parent_id=0, is_active=True)
        if self.value() == 0:
            return qs
        else:
            qs = qs.filter(project_id=self.value())
```
In my Admin.py 
```
list_filter = ['status', ProjectFilter]
suit_list_filter_horizontal = list_filter
```
but the project show like this:
![](https://test-media-cmcaifu-com.oss-cn-hangzhou.aliyuncs.com/upload/a0ceae4f40974d03924e6bbd88d07570.jpg)

I want it show on top， how to.

**Comment by Datajuristes** [comment]
Hi thanks for your work, i love django suit
I juste have a question: why don't you make the django suit admin responsive ? It might be possible with bootstrap

**Comment by bskim45** [comment]
@Datajuristes its in TODO list

**Comment by larryhq** [comment]
suit2.0 not support django2.0

**Comment by jrthorne** [comment]
The documentation on configuration is pretty sparse. In the end trial and error and following 
https://docs.djangoproject.com/en/1.11/ref/contrib/admin/#adminsite-attributes
https://docs.djangoproject.com/en/2.0/ref/applications/

Essentially, you need to have an 'apps.py' file containing the config, 

`lass SuitConfig(DjangoSuitConfig):
    layout = 'horizontal'
    verbose_name = 'Introtravel Shopping Cart'
    menu = (
        ParentItem('Batch Processes', children=[
            ChildItem('Queued Tasks', url='/backend/common/queuedtask/'),
            ChildItem('Cron Tasks', url='/backend/common/crontask/'),
        ], icon='fa fa-leaf'),
        ParentItem('Money', children=[
            ChildItem('Transactions', url='/backend/products/transaction/'),
            ChildItem('Paypal Records', url='/backend/pro/paypalnvp/'),
            ChildItem('Prices', url='/backend/products/price/')
        ], icon='fa fa-money'),
        ParentItem('Products', children=[
            ChildItem('Bookings', url='/backend/products/booking/'),
            ChildItem('Tours', url='/backend/products/basetour/')
        ],icon='fa fa-flag'),
        ParentItem('Users', children=[
            ChildItem(model='auth.user'),
            ChildItem('User groups', 'auth.group'),
            ChildItem('Guests', url='/backend/common/guest/')
        ], icon='fa fa-users'),
        ParentItem('Right Side Menu', children=[
            ChildItem('Password change', url='admin:password_change'),
            ChildItem('Open Help Doc', url='https://docs.google.com/document/d/1o-_pdaMYLexnyhmBaKsVXp1L_5tq1dSONExPijH09X0/edit', target_blank=True),

        ], align_right=True, icon='fa fa-cog'),
    )

    admin_name = 'Cart'`

The in your settings. py `
INSTALLED_APPS = [...
'common.apps.SuitConfig',...
]`

**Comment by novelview9** [comment]
Thanks all about awesome v2 feature!
I wonder fieldsets options full-width on v2

**Comment by cometsong** [comment]
@larryhq The suit_list_filter-horizontal uses the `parameter_name` or field name, not the Filter class name.  As shown in issue:  #592

**Comment by larryhq** [comment]
@cometsong Thank you very much.

**Comment by wanghepeng1986** [comment]
when can suit2.0  support django2.0 ？

**Comment by kamilgregorczyk** [comment]
Yeah, ModuleNotFoundError: No module named 'django.core.urlresolvers' on django 2.0

**Comment by Arlington1985** [comment]
is it possible to create menu parentitem without childitem?

**Comment by larryhq** [comment]
@Arlington1985  use app can without childitem, like this:
```
ParentItem(_('accounts'), app='accounts', icon='fa fa-user'),
```

**Comment by larryhq** [comment]
Hi, I'm defining my suit config as the following and those models exists but the second menu don't show, it just shows the first one link. I don't know what I'm duing wrong.
```
ParentItem(_('钰翔提单'), children=[ChildItem(model='yx_orders.delivery'),
                                        ChildItem(model='yx_orders.deliveryorder')], icon='fa fa-leaf'),
```

**Comment by Arlington1985** [comment]
@larryhq, maybe I haven't wrote clearly, but I want to create a menu item without having sub-menu. Your suggestion lists all the models under the app as a submenu accounts app. I would like to show user model only as a standalone menu item, **without being a child** any parent(`accounts`) menu

**Comment by larryhq** [comment]
@Arlington1985 
```
ParentItem(_('test'), url='/admin/accounts/', icon='fa fa-leaf')
```
like this?

**Comment by Arlington1985** [comment]
@larryhq, thanks.
This is what I want. So it's only possible with url.

**Comment by larryhq** [comment]
@Arlington1985 You're welcome! 😄

**Comment by larryhq** [comment]
Can I custom multiple search fields in admin like this?
![](https://test-media-cmcaifu-com.oss-cn-hangzhou.aliyuncs.com/upload/22a2d1b6c167464f9140c3b34ee8875d.png)

**Comment by larryhq** [comment]
Can I filter queryset from start_time to end_time?

**Comment by george72** [comment]
Any news on the v2 release date? is it Django 2.0 compatible?

**Comment by SMaciej** [comment]
Great to see that you are working on a new version! :+1: 
Do you considered mobile devices. For now on smartphone resolutions vertical menu covers the content, the horizontal one will take too much space, if there are lots of elements.

**Comment by gamesbook** [comment]
Re an earlier comment by @darklow about altering/overriding the `base_site.html` file.

I have tried copying the `suit/admin/base.html` as a `base_site.html` file - first directly into my project `templates` directory and then under the `templates/admin` directory, but in neither case did any changes I made in that `base_site.html` file show up on my site.  What is the correct way to do this?

**Comment by jdruiter** [comment]
Great to see more potential compatability with third party django apps. I do like the previous css better and haven't found any reason to upgrade. I'm sticking with the v1.

**Comment by eborrell** [comment]
@darklow we are already using v2 on production. The 'LinkedSelect' feature and links to foreign keys and second order foreign keys on the admin list are great. However, we are not able to put the actions on top of the list view. 
    
    ModelAdmin.actions_on_top = True

does not work anymore.

**Comment by gdreich** [comment]
The new version looks great! I was wondering if there is anyway to implement multiple admin sites using v2?

**Comment by wdschn** [comment]
Lost style in your demo.

**Comment by odwrotnie** [comment]
What about the Responsiveness and Bootstrap production version?

**Comment by kamilgregorczyk** [comment]
Will this version of Django suit ever be released?

niedz., 29 lip 2018, 10:28 użytkownik Derek Hohls <notifications@github.com>
napisał:

**Comment by ArnoldGitHub** [comment]
The last time that the project owner @darklow post was nearly a year ago, 10 Sep 2017. I have no choice but to use the original admin that comes with Django version 2.

**Comment by kamilgregorczyk** [comment]
It's responsive now and not so bad as it used to be.

niedz., 5 sie 2018, 16:50 użytkownik Arnold Chan <notifications@github.com>
napisał:

**Comment by ArnoldGitHub** [comment]
Yes, the original admin of Django is never bad too use. Just the Django Suit is damn cool, especially v2.
It is sad that Django Suit doesn’t keep updating.

I am a paid user of Django Suit.

**Comment by gamesbook** [comment]
@ArnoldGitHub That is not strictly true; @darklow did a V2 release in February this year -see https://github.com/darklow/django-suit/releases/tag/v.0.2.26

But I agree that (as a paid user) I would like to see a formal release of v2.

**Comment by kamilgregorczyk** [comment]
Still, it's August

niedz., 5 sie 2018 o 17:41 Derek Hohls <notifications@github.com>
napisał(a):

**Comment by ArnoldGitHub** [comment]
@gamesbook Although, version 0.2.26 is compatible with Django 2.0, it is Django Suit v1, not Django Suit v2.0 of this thread

**Comment by EOSIT** [comment]
@ArnoldGitHub Am I missing something - the very first commit in that series (https://github.com/darklow/django-suit/compare/v.0.2.26...v2) called:

> Fresh start - v2.0a1

**Comment by ArnoldGitHub** [comment]
Sorry, my fault, i incorrectly used "pip install django-suit", instead of the correct "pip install https://github.com/darklow/django-suit/tarball/v2"

**Comment by majbah08** [comment]
tables are not responsive, how can solve it for v2 . thanks in advance

**Comment by tjerkw** [comment]
Whats the status of this? Should we drop django suits?

**Comment by kamilgregorczyk** [comment]
It looks abandoned

**Comment by SalahAdDin** [comment]
Yeah, it looks like.

**Comment by willseward** [comment]
ping @darklow 

Is this abandoned?

**Comment by darklow** [comment]
I use Django Suit (v2-dev version) for all my projects every day, however I have no time working on new features, yet I am using Django suit v2-dev on 10+ production projects without any issues with latest Django 2.x versions. 

So project is not abandoned, and all critical pull requests have been approved, however new feature development is not planned in nearest time nor I have time to write docs for v2. Remember that this is just a skin on top of awesome Django admin app and you can easily extend things with template override or css or js.

**Comment by ebertti** [comment]
I use v2-dev version too.

It works fine for me.

**Comment by gamesbook** [comment]
@darklow Despite those limits it would be really great to have a PyPi release (the last was over a year ago).

**Comment by darklow** [comment]
@gamesbook My only worry releasing pypi version of v2.x is that there are no docs for v2 and users are used to upgrade without looking at versions much and installing v2 over old v1 may break their apps and there are even no docs to see how to properly use v2 except the demo app.

**Comment by gamesbook** [comment]
@darklow It could be released as an "Alpha" with the README upfront saying **"NO DOCS. DO NOT USE TO UPGRADE FROM v1".**  I think developers are a bit smarter than just blindly running an "upgrade".

**Comment by diegoasencio96** [comment]
> ¿Alguien que ya usa v2.0 alpha en desarrollo o es solo yo? :)

Yoooo también la uso y me parece suprema mente espectacular!!!. Saludos desde Colombia, suit v2 es un gran aporte !!! felicitaciones!!! realmente admirado con este trabajo que cada día se hace mas elegante.

**Comment by diegoasencio96** [comment]
> @gamesbook Mi única preocupación al lanzar la versión pypi de v2.x es que no hay documentos para v2 y los usuarios están acostumbrados a actualizar sin mirar mucho las versiones y la instalación de v2 sobre la v1 anterior puede romper sus aplicaciones e incluso no hay documentos para ver cómo para utilizar correctamente v2, excepto la aplicación de demostración.

Con  la demo y un poco de astusia, es suficiente jejeje

**Comment by tjerkw** [comment]
@darklow  +1 for releasing this as an alpha!

**Comment by Lessica** [comment]
Django Suit is great! I used 2.0 in my open-source project [DCRM](https://github.com/82Flex/DCRM).

**Comment by tyctor** [comment]
> I use Django Suit (v2-dev version) for all my projects every day, however I have no time working on new features, yet I am using Django suit v2-dev on 10+ production projects without any issues with latest Django 2.x versions.

@darklow you probably meant Django 2.0.x versions, because from Django 2.1. there is jQuery 3.3.1 (instead of previous 2.2.3), which will not work with `suit.sortables.js` because  `$(this).selector` is not supported anymore
so note for everyone who want to use Django 2.1.x or higher, your SortableAdmin wil not work due to that change

**Comment by madthew** [comment]
will there be a Django 3 compatible version?

**Comment by geceba** [comment]
@madthew At the moment I have not found a problem
Related fix: #711

**Comment by dlooto** [comment]
Why not support Django3.0 and Python 3.8+ ? Thanks~

**Comment by darklow** [comment]
> Why not support Django3.0 and Python 3.8+ ? Thanks~

Have you tried? I'm using v2 with Django 3.0 and Python 3.6.4 without issues.

**Comment by niteshrawat1995** [comment]
Hi @darklow  can we implement the apps reorder functionality with v2 ? I am talking about this: https://django-suit.readthedocs.io/en/latest/configuration.html.

**Comment by darklow** [comment]
> Hi @darklow can we implement the apps reorder functionality with v2 ? I am talking about this: https://django-suit.readthedocs.io/en/latest/configuration.html.

You can already do this, see v2 demo app and its code:
https://github.com/darklow/django-suit/blob/v2/demo/demo/apps.py

**Comment by niteshrawat1995** [comment]
Hi @darklow , price schemes of v2 is same as v1 right ?

**Comment by niteshrawat1995** [comment]
@darklow bro how can i override the templates like base_site.html , login.html of admin with your changes intact. I tries to directly override them  via templates/admin/base_site.html but it removed your changes which were overlapping.

**Comment by darklow** [comment]
> Hi @darklow , price schemes of v2 is same as v1 right ?
Yes, and v1 licenses are valid for v2 too.

> @darklow bro how can i override the templates like base_site.html , login.html of admin with your changes intact. I tries to directly override them via templates/admin/base_site.html but it removed your changes which were overlapping.

See example in demo app:
https://github.com/darklow/django-suit/blob/v2/demo/demo/templates/admin/base_site.html

**Comment by pablovg2** [comment]
Hi, I am using this version but I have a problem with popup they are open in the same windows without popup

**Comment by pierre-H** [comment]
Any news ?

**Comment by paulocoutinhox** [comment]
It still active?

**Comment by madthew** [comment]
> It still active?
It's still good to be used. but if you need some specific features of Django 3.x or you hope to use something maintained I think the answer is not.

**Comment by pulse-mind** [comment]
It it can helps, You can have a look to the forks, at least mine is using django v3 for example.

**Comment by SalahAdDin** [comment]
Isn't it too late for this one?

**Comment by tjerkw** [comment]
You can consider this project abandoned, don't use it for new projects.
The last update to the readme was from 2016!!!

 We currently heavily depend on it, but it blocks us from upgrading django. Django's architecture is heavily depending on dynamic typing which makes it really hard to check if things are compatible. We run into runtime issues constantly.

**Comment by pulse-mind** [comment]
I am using my fork on django V3 and it works fine but not yet on prod. 
If you like you can participate to mine.

**Comment by mhindery** [comment]
I am using the latest Django (v3.2.4) with Django-suit v1 currently in production. We also have a heavy dependancy on it, so we can't easily switch to a different package, or to v2 for that matter. My changes are on https://github.com/mhindery/django-suit/tree/develop should it be useful to you.

**Comment by iflare3g** [comment]
@mhindery thanks so much! Which are your plans ? Will you maintain your fork ? 
Django-suit is a heavy dependency also inside my project and it's a blocker for the Django 3 migration.
Obvs we can remove it and use the default Django Admin but we have to refactor and adjust every custom stuff we have built upon Django Suit

**Comment by mhindery** [comment]
We are in the same situation as you that the default Django admin is insufficient and we can't just refactor everything to a new package or version. I'm maintaining it in the sense that we run the latest Django at the company where I work, and if we notice something doesn't work anymore during upgrades, I fix it. There are probably things that we aren't using from django-suit which I thus cannot comment on that might be broken. I'm not building any new functionality in there. If you need both django-suit v1 and an uptodate Django version, you should be good with it.

**Comment by gamesbook** [comment]
> I am using my fork on django V3 and it works fine but not yet on prod.
> If you like you can participate to mine.

Is this a django-suit v2 fork? I very much doubt the original developer will ever return to v1 so to continue trying to update that is effectively going down a dead end (and it will eventually become unusable).

**Comment by iflare3g** [comment]
yeah I've just tested your fork and it unblocks my migration to Django 3.x so it's perfect so far ( thanks so much!! )
my "fear" is that when a new LTS will drop, we will go back to the same situation so at that point it's better to remove it at all and start crying by refactoring inside the default admin panel 😞

**Comment by mhindery** [comment]
Mine is a v1 fork. I agree it is not a long-term solution and at some point in the future we'll have to re-evaluate it. However with the recent Django upgrades (2.x and 3.x) there hasn't been massive changes in the admin that needed big refactorings in django-suit; we aren't missing functionality from v2 and are happy with what v1 does, so it's just not worth investing into something different at this time.

**Comment by mhindery** [comment]
Mine is a v1 fork. I agree it is not a long-term solution and at some point in the future we'll have to re-evaluate it. However with the recent Django upgrades (2.x and 3.x) there hasn't been massive changes in the admin that needed big refactorings in django-suit; we aren't missing functionality from v2 and are happy with what v1 does, so it's just not worth investing into something different at this time.

**Comment by iflare3g** [comment]
@mhindery yup, agree with u!

**Comment by g2-im** [comment]
Hi,
I have install and configured. Then copied over template and css folders to my project. routing through base.html.
Recent install has no code in files such as dashboard, change_list, and others.
After logging into django admin. New top menu interface looks great. But as I click on any link say users... 
the display gets disturbed and link contents floats to downside of the page and can not click on add user or any item. It seems bootstrap css issue but I gave up  #...
![Screenshot from 2021-10-04 01-06-44](https://user-images.githubusercontent.com/89628959/135757346-7ef9d76d-bc2a-4f3b-8e03-8bed5bf18a45.png)
not sure how to fix it.

**Comment by g2-im** [comment]
Content >> Custom View 
throws:

<h1>Page not found <span>(404)</span></h1>  

Request Method: | GET
-- | --
http://127.0.0.1:8000/admin/custom/
django.contrib.admin.sites.catch_all_view
<list of all registered views tried but found >
     admin/ (?P<url>.*)$

The current path, admin/custom/, matched the last one.

**Comment by g2-im** [comment]
Content >> Custom view
of all registers admin views throws 404
<h1>Page not found <span>(404)</span></h1>
    
 Request Method: | GET
-- | --
http://127.0.0.1:8000/admin/custom/
django.contrib.admin.sites.catch_all_view

Using the URLconf defined in jvRTOms.urls, Django tried these URL patterns, in this order: 

  admin/ (?P<url>.*)$

The current path, admin/custom/, matched the last one.

**Comment by g2-im** [comment]
My app folder __pycache__
shows files such as __init__.cpython-38.pyc

**Comment by g2-im** [comment]
My app folder "__pycache__"
shows files such as "__init__.cpython-38.pyc"

**Comment by g2-im** [comment]
Can someone help me with it??

**Comment by jorenham** [comment]
@g2-im This repo appears to be abandoned. But there is fork of v2 you could try: https://github.com/pulse-mind/django-suit/tree/v2

**Comment by g2-im** [comment]
> @g2-im This repo appears to be abandoned. But there is fork of v2 you could try: https://github.com/pulse-mind/django-suit/tree/v2

Hi Mate,

I am new to Django. Can you direct me to about how to setup these projects and run on my PC.
Normally I am missing how to use the setup file in your codes that I downloaded from your link you advised.

Because if I can setup your project and run it then good platform for me to learn and fix some of issues.

Any youtube link or article can help me please.

Regards,

**Comment by g2-im** [comment]
I just learned from youtube.... :)
pip install -r requirements.txt
and 
python setup.py develop

it links suit with demo ...;)
